contracts/clearing/TradingApp.sol

```diff
@@ -37,6 +37,28 @@ interface ITradingStructs {
 
 // FIXME: should Vault support multiple brokers?
 interface ISettle {
+	// ========== Events ==========
+
+	/**
+	 * @notice Emitted when a channel is settled.
+	 * @param trader The address of the trader.
+	 * @param broker The address of the broker.
+	 * @param channelId The ID of the channel.
+	 */
+	event Settled(address indexed trader, address indexed broker, bytes32 indexed channelId);
+
+	// ========== Errors ==========
+
+	error InvalidStateTransition(string reason);
+
+	// ========== Functions ==========
+
+	/**
+	 * @notice Settle a channel.
+	 * @param fixedPart The fixed part of the state.
+	 * @param proof The proof of the state.
+	 * @param candidate The candidate state.
+	 */
 	function settle(
 		INitroTypes.FixedPart calldata fixedPart,
 		INitroTypes.RecoveredVariablePart[] calldata proof,
```

Comment:
For some reason I cannot comment the code below this line
For settlement I would still verify order, trade and that turnNumber are consecutive.

Reply:
> For settlement I would still verify order, trade and that turnNumber are consecutive.

In the `TradingApp` nitro app we require that the settlement is signed by both parties, which means that we CAN outsource the validation of orders, trades and turn numbers to them for SIMPLICITY. Having said that, do you want to explicitly validate the settlement in the nitro app? This will require supplying all order-trade pairs, which can take a lot of block space, incurring high transaction fees.

Reply:
Best way to answer this would be to measure the additional costs, but yes, I would still provide them at and validate them at this stage.

it's better that both backend, and blockchain validate the same rules.